### PR TITLE
more testing for mtfnam

### PR DESCRIPTION
**Pull request by jbathegit**
Part of #445

### CODE REVIEW

File: src/mtinfo.f [hunk 1]
```diff
@@ -72,6 +70,4 @@ RECURSIVE SUBROUTINE MTINFO ( CMTDIR, LUNMT1, LUNMT2 )
 C  -----
 
       RETURN
-900   BORT_STR = 'BUFRLIB: MTINFO - BAD INPUT MASTER TABLE DIRECTORY:'
```
Comment (AlexanderRichert-NOAA) [inline src/mtinfo.f:75]: Is removing these lines related to testing or something else?

Comment (jbathegit) [reply src/mtinfo.f:75]: I was trying to test for this in test_bort.  But I quickly realized that, with the changes to strsuc in #406, it's no longer possible for that routine to return with LMTD<0, which in turn means that lines 75 and 76 can no longer be reached within this routine.

I could have changed the if test to LMTD<=0, but I realized that whatever values get returned for MTDIR and LMTD will always be quickly checked anyway (via an inquire statement inside of subroutine mtfnam), so this check in strsuc is now basically superfluous and can be removed.

**Review by AlexanderRichert-NOAA — APPROVED** [review]

**Comment by jack-woollen** [comment]
Jeff

I just got this from cloning/compiling a bufrlib branch on wc2.

/lfs/h2/emc/global/noscrub/Jack.Woollen/cris-fix/NCEPLIBS-bufr/test/test_misc.F90(207):
error #6345: Either a PAUSE, STOP or ERROR STOP statement has an invalid
argument.
    if ( (iret .ne. 0) .or. (igetprm(trim(prms(i))) .ne. 42+i) ) stop 800+i

It doesn't like stop 800+i !

Jack

**Comment by AlexanderRichert-NOAA** [comment]
Ah dang it, that was me. I forgot some compilers/fortran versions don't like that. I'm looking at this file right now so I'll fix it while I'm at it.

**Comment by jbathegit** [comment]
Hey Jack, just curious - what compiler were you trying to use on WC2?  If it was the Intel compiler, then it's surprising that this hasn't already cropped up during the GitHub CI runner testing.

In short, I agree we need to get this fixed, but just wondering why it never showed up in the CI test for Intel?  More of a curiosity as much as anything.

**Pull request merged**